examples/foundational/21-tavus-layer.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (c) 2024, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+import asyncio
+import aiohttp
+from typing import Any
+import os
+import sys
+
+from pipecat.frames.frames import LLMMessagesFrame
+from pipecat.pipeline.pipeline import Pipeline
+from pipecat.pipeline.runner import PipelineRunner
+from pipecat.pipeline.task import PipelineParams, PipelineTask
+from pipecat.processors.aggregators.llm_response import (
+    LLMAssistantResponseAggregator,
+    LLMUserResponseAggregator,
+)
+from pipecat.services.cartesia import CartesiaTTSService
+from pipecat.services.openai import OpenAILLMService
+from pipecat.services.deepgram import DeepgramSTTService
+from pipecat.services.tavus import TavusVideoService
+from pipecat.transports.services.daily import DailyParams, DailyTransport
+from pipecat.audio.vad.silero import SileroVADAnalyzer
+
+from loguru import logger
+
+from dotenv import load_dotenv
+
+load_dotenv(override=True)
+
+logger.remove(0)
+logger.add(sys.stderr, level="DEBUG")
+
+
+async def main():
+    try:
+        async with aiohttp.ClientSession() as session:
+            # get persona, look up persona_name, set this as the bot name to ignore
+            persona_name = await TavusVideoService.get_persona_name(
+                session=session,
+                api_key=os.getenv("TAVUS_API_KEY"),
+                persona_id=os.getenv("TAVUS_PERSONA_ID", "pipecat0"),
+            )
+
+            room_url, conversation_id = await TavusVideoService.initiate_conversation(
+                session=session,
+                api_key=os.getenv("TAVUS_API_KEY"),
+                replica_id=os.getenv("TAVUS_REPLICA_ID"),
+                persona_id=os.getenv("TAVUS_PERSONA_ID", "pipecat0"),
+            )
```

Comment:
This would go away if you follow the other comment.

Reply:
I just realized we need the `room_url`... I think some of it still applies. Instead of `StartFrame` you could do it in a function:

```
tavus = TavusVideoService(
    api_key=os.getenv("TAVUS_API_KEY"),
    replica_id=os.getenv("TAVUS_REPLICA_ID"),
    persona_id=os.getenv("TAVUS_PERSONA_ID", "pipecat0")
    aiohttp_session=session)

persona_name = await tavus.get_persona_name()

room_url = await tavis.initialize()
```